Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Sliders/PinchSlider.cs

```diff
@@ -48,17 +48,87 @@ public float SliderValue
             }
         }
 
+        [Header("Slider Axis Visuals")]
+
+        [Tooltip("The gameObject that contains the thumb visual. This will get rotated to match the slider axis")]
+        [SerializeField]
+        private GameObject thumbVisual = null;
+        public GameObject ThumbVisual
+        {
+            get
+            {
+                return thumbVisual;
+            }
+            set
+            {
+                thumbVisual = value;
+                UpdateThumbVisuals();
+            }
+        }
+    
+        [Tooltip("The gameObject that contains the trackVisuals. This will get rotated to match the slider axis")]
+        [SerializeField]
+        private GameObject trackVisuals = null;
+        public GameObject TrackVisuals
+        {
+            get
+            {
+                return trackVisuals;
+            }
+            set
+            {
+                trackVisuals = value;
+                UpdateTrackVisuals();
+            }
+        }
+
+        [Tooltip("The gameObject that contains the tickMarks.  This will get rotated to match the slider axis")]
+        [SerializeField]
+        private GameObject tickMarks = null;
+        public GameObject TickMarks
+        {
+            get
+            {
+                return tickMarks;
+            }
+            set
+            {
+                tickMarks = value;
+                UpdateTickMarks();
+            }
+        }
+
+
         [Header("Slider Track")]
 
         [Tooltip("The axis the slider moves along")]
         [SerializeField]
         private SliderAxis sliderAxis = SliderAxis.XAxis;
-        [Serializable]
-        private enum SliderAxis
+        public SliderAxis CurrentSliderAxis
         {
-            XAxis = 0,
-            YAxis,
-            ZAxis
+            get { return sliderAxis; }
+            set
+            {
+                sliderAxis = value;
+                UpdateVisualsOrientation();
+            }
+        }
+
+        private SliderAxis? previousSliderAxis = null;
+        public SliderAxis PreviousSliderAxis
```

Comment:
Does this need to be `public`? It seems like local state management that we wouldn't necessarily want other scripts arbitrarily setting.

Reply:
I ve changed it to private.